src/ir/ty.rs

```diff
@@ -365,6 +353,16 @@ impl IsOpaque for Type {
     }
 }
 
+/// Whether this type has a vtable.
+impl HasVtable for Type {
+    // Note that we need the Type's Item to answer this question.
+    type Extra = ItemId;
+
+    fn has_vtable(&self, ctx: &BindgenContext, itemid: &ItemId) -> bool {
+        ctx.lookup_item_id_has_vtable(itemid)
+    }
```

Comment:
I think the ItemId and Item one doesn't get used. The has_vtable is called on Ty in the codegen/mod.rs inside the impl CodeGenerator for CompInfo. needs_explicit_vtable in CompInfo and is_unsized in Ty and CompInfo also call the has_vtable. I think for Ty and CompInfo, calling lookup_item_id_has_vtable directly might be better. But if we want to keep the same interface for all the types, I'm also fine with that